app/models/user.rb

```diff
@@ -46,6 +48,29 @@ def events
     accepted_applications.collect { |a| a.event }
   end
 
+  # Returns true iff. user has submitted an agreement_letter for the given event
+  def agreement_letter_for_event?(given_event)
+    fitting_agreement_letters = self.agreement_letters.select { |letter| letter.event == given_event }
+	return false unless fitting_agreement_letters.length == 1
+	return true
+  end
+  
+  # Returns the agreement letter the user has submitted for given_event. Returns false if no such letter exists.
+  def agreement_letter_for_event(given_event)
+    fitting_agreement_letters = self.agreement_letters.select { |letter| letter.event == given_event }
+	return false unless fitting_agreement_letters.length == 1
```

Comment:
If the caller is expecting an AgreementLetter you shouldn't return a Boolean. You can remove the this line and just return `fitting_agreement_letters[0]`. If no AgreementLetter exists, it will return nil.